homeassistant/components/binary_sensor/flic.py

```diff
@@ -0,0 +1,240 @@
+    double_click_threshold = config.get(CONF_THRESHOLD_DOUBLE_CLICK)
+    long_click_threshold = config.get(CONF_THRESHOLD_LONG_CLICK)
+
+    button = FlicButton(hass, client, address,
+                        double_click_threshold, long_click_threshold)
+    _LOGGER.info("Connected to button (%s)", address)
+
+    yield from async_add_entities([button])
+
+
+@asyncio.coroutine
+def async_get_verified_addresses(client):
+    """Retrieve addresses of verified buttons."""
+    future = asyncio.Future()
+    loop = asyncio.get_event_loop()
+
+    def get_info_callback(items):
+        """Set the addressed of connected buttons as result of the future."""
+        addresses = items["bd_addr_of_verified_buttons"]
+        run_callback_threadsafe(loop, future.set_result, addresses)
+    client.get_info(get_info_callback)
+
+    return future
+
+
+class FlicButton(BinarySensorDevice):
+    """Representation of a flic button."""
+
+    def __init__(self, hass, client, address,
+                 double_click_threshold, long_click_threshold):
+        """Initialize the flic button."""
+        import pyflic
+
+        self._hass = hass
+        self._address = address
+        self._double_click_threshold = double_click_threshold
+        self._long_click_threshold = long_click_threshold
+        self._is_down = False
+        self._last_click = self._last_down = self._last_up = datetime.min
+
+        # Initialize connection channel
+        self._channel = pyflic.ButtonConnectionChannel(self._address)
+        self._channel.on_button_up_or_down = self._button_up_down
+        client.add_connection_channel(self._channel)
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self.address.replace(":", "")
+
+    @property
+    def address(self):
+        """Return the bluetooth address of the device."""
+        return self._address
+
+    @property
+    def is_on(self):
+        """Return true if sensor is on."""
+        return self._is_down
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return False
+
+    @property
+    def state_attributes(self):
+        """Return device specific state attributes."""
+        attr = super(FlicButton, self).state_attributes
+        attr["address"] = self.address
+
+        return attr
+
+    def _button_up_down(self, channel, click_type, was_queued, time_diff):
+        """Fire click event depending on click type."""
+        import pyflic
+
+        if not was_queued:
+            # Set state
+            self._is_down = click_type == pyflic.ClickType.ButtonDown
+            self.update_ha_state()
+
+            # Fire appropriate event
+            now = datetime.now()
+            last_down_diff = (now - self._last_down).total_seconds()
+            last_click_diff = (now - self._last_click).total_seconds()
+
+            if self._is_down:
+                self._last_down = now
+            else:
```

Comment:
Instead of this `else`, include a `return` in the `if` statement above and you don't have to indent the other code that much